app/src/main/java/com/keylesspalace/tusky/components/timeline/viewmodel/NetworkTimelineViewModel.kt

```diff
@@ -145,28 +150,46 @@ class NetworkTimelineViewModel @Inject constructor(
                     return@launch
                 }
 
-                val activeAccount = accountManager.activeAccount!!
+                statusData.removeAt(placeholderIndex)
 
+                val activeAccount = accountManager.activeAccount!!
                 val data = statuses.map { status ->
-                    val oldStatus = statusData.find { s ->
-                        s.asStatusOrNull()?.id == status.id
-                    }?.asStatusOrNull()
-
-                    val contentShowing = oldStatus?.isShowingContent ?: activeAccount.alwaysShowSensitiveMedia || !status.actionableStatus.sensitive
-                    val expanded = oldStatus?.isExpanded ?: activeAccount.alwaysOpenSpoiler
-                    val contentCollapsed = oldStatus?.isCollapsed ?: true
-
                     status.toViewData(
-                        isShowingContent = contentShowing,
-                        isExpanded = expanded,
-                        isCollapsed = contentCollapsed
+                        isShowingContent = activeAccount.alwaysShowSensitiveMedia || !status.actionableStatus.sensitive,
+                        isExpanded = activeAccount.alwaysOpenSpoiler,
+                        isCollapsed = true
                     )
+                }.toMutableList()
+
+                if (statuses.isNotEmpty()) {
+                    val firstId = statuses.first().id.hashCode().toLong()
+                    val lastId = statuses.last().id.hashCode().toLong()
+                    val overlappedFrom = statusData.indexOfFirst { it.viewDataId <= firstId }
+                    val overlappedTo = statusData.indexOfFirst { it.viewDataId < lastId }
+
+                    if (overlappedFrom < overlappedTo) {
+                        repeat(overlappedTo - overlappedFrom) {
+                            statusData[overlappedFrom].asStatusOrNull()?.let { oldStatus ->
+                                val dataIndex = statuses.indexOfFirst { it.id == oldStatus.id }
+                                if (dataIndex == -1) {
+                                    return@let
+                                }
+                                data[dataIndex] = data[dataIndex]
+                                    .copy(
+                                        isShowingContent = oldStatus.isShowingContent,
+                                        isExpanded = oldStatus.isExpanded,
+                                        isCollapsed = oldStatus.isCollapsed,
+                                    )
+                            }
+
+                            statusData.removeAt(overlappedFrom)
```

Comment:
Why is it not enough to insert this somewhere?
```kotlin
if (oldStatus != null) {
     statusData.remove(oldStatus)
}
```

Reply:
If some statuses in overlapped range removed by author, they could be correctly removed by this process.
I implemented this behaviour because I thought that we should remove statuses which we could know they are removed.
If you think we don't need this behaviour, I can remove this.

Reply:
Ah I see, yes we should have this if possible.

But I think there are two problems with your code
1) bothering with list indizes makes it hard to understand
2) `hashcode()` ordering does not equal object ordering, we must compare by actual status id length & order

Here is what I would try to do

```kotlin
if (statuses.isNotEmpty()) {
    val firstId = statuses.first().id
    val lastId = statuses.last().id
    statusData.removeAll { status ->
        val s = status.asPlaceholderOrNull()
        s != null && s.id.isLessOrEqualTo(firstId) && s.id.isMoreOrEqualTo(lastId)
    }
}
```
where `isLessOrEqualTo` and `isMoreOrEqualTo` are custom extension functions similar to [this one](https://github.com/tuskyapp/Tusky/blob/b145fc9d506875b7d1ad6f2786d6c3c6708b7407/app/src/main/java/com/keylesspalace/tusky/util/StringUtils.kt#L73)

Reply:
I improved the logic bit. Also stopped using `hashCode` for ordering.